docs/pin.rst

```diff
@@ -0,0 +1,172 @@
+|  2  | Touch   | Pad 2    |
++-----+---------+----------+
+|  3  | Analog  | Column 1 |
++-----+---------+----------+
+|  4  | Digital | Button A |
++-----+---------+----------+
+|  5  | Analog  | Column 2 |
++-----+---------+----------+
+|  6  | Digital | Row 2    |
++-----+---------+----------+
+|  7  | Digital | Row 1    |
++-----+---------+----------+
+|  8  | Digital |          |
++-----+---------+----------+
+|  9  | Digital | Row 3    |
++-----+---------+----------+
+|  10 | Analog  | Column 3 |
++-----+---------+----------+
+|  11 | Digital | Button B |
++-----+---------+----------+
+|  12 | Digital |          |
++-----+---------+----------+
+|  13 | Digital | SPI MOSI |
++-----+---------+----------+
+|  14 | Digital | SPI MISO |
++-----+---------+----------+
+|  15 | Digital | SPI SCK  |
++-----+---------+----------+
+|  16 | Digital |          |
++-----+---------+----------+
++-----+---------+----------+
+|  19 | Digital | I2C SCL  |
++-----+---------+----------+
+|  20 | Digital | I2C SDA  |
++-----+---------+----------+
+
+The above table summarizes the pins available, their types (see below) and what
+they are internally connected to.
+
+
+Pulse-Width Modulation
+----------------------
+
+The pins of your board cannot output analog signal the way an audio amplifier
+can do it -- by modulating the voltage on the pin. Those pins can only either
+enable the full 3.3V output, or pull it down to 0V. However, it is still
+possible to control the brightness of LEDs or speed of an electric motor, by
+switching that voltage on and off very fast, and controlling how long it is on
+and how long it is off. This technique is called Pulse-Width Modulation (PWM),
+and that's what the ``write_analog`` method below does.
+
+.. image:: pwm.png
+
+Above you can see the diagrams of three different PWM signals. All of them have
+the same period (and thus frequency), but they have different duty cycles.
+
+The first one would be generated by ``write_analog(511)``, as it has exactly
+50% duty -- the power is on half of the time, and off half of the time. The
+result of that is that the total energy of this signal is the same, as if it
+was 1.65V instead of 3.3V.
+
+The second signal has 25% duty cycle, and could be generated with
+``write_analog(255)``. It has similar effect as if 0.825V was being output on
+that pin.
+
+The third signal has 75% duty cycle, and can be generated with
+``write_analog(767)``. It has three times as much energy, as the second signal,
+and is equivalent to outputting 2.475V on th pin.
+
+Note that this works well with devices such as motors, which have huge inertia
+by themselves, or LEDs, which blink too fast for the human eye to see the
+difference, but will not work so good with generating sound waves. This board
+can only generate square wave sounds on itself, which sound pretty much like
+the very old computer games -- mostly because those games also only could do
+that.
+
+
+Hobby Servomechanisms
+---------------------
```

Comment:
I was just about to suggest the same.

Is there going to be example documentation stored in this repository as well then? It might be hard to define a scope of how many examples might be needed or what would be outside of the scope of the documentation.